src/Languages/JavaScript.ts

```diff
@@ -0,0 +1,403 @@
+     */
+    protected generateClassMemberVariableProperties(variables: ClassMemberVariableProperties): void {
+        super.generateClassMemberVariableProperties(variables);
+
+        variables.protectedCase = CaseStyle.CamelCase;
+        variables.publicCase = CaseStyle.CamelCase;
+    }
+
+    /**
+     * Generates metadata on comments.
+     * 
+     * @param comments   A property container for metadata on comments.
+     */
+    protected generateCommentProperties(comments: CommentProperties): void {
+        super.generateCommentProperties(comments);
+
+        comments.docEnd = " */";
+        comments.docLineEnd = "";
+        comments.docLineStart = " * ";
+        comments.docTagAliases = {
+            "note": "remarks",
+            "parameter": "param",
+            "returns": "returns",
+            "summary": "",
+            "todo": "todo"
+        };
+        comments.docTagsWithParameters = {
+            "summary": "\0",
+            "parameter": ""
+        };
+        comments.docTagEnd = " ";
+        comments.docTagSpaceAfter = "  ";
+        comments.docTagStart = "@";
+        comments.docStart = "/**";
+    }
+
+    /**
+     * Generates metadata on conditionals.
+     * 
+     * @param conditionals   A property container for metadata on conditionals.
+     */
+    protected generateConditionalProperties(conditionals: ConditionalProperties): void {
+        super.generateConditionalProperties(conditionals);
+
+        conditionals.continueLeft = "} ";
+        conditionals.continueRight = " {";
+        conditionals.startRight = ") {";
+    }
+
+    /**
+     * Generates metadata on dictionaries.
+     * 
+     * @param dictionaries   A property container for metadata on dictionaries.
+     */
+    protected generateDictionaryProperties(dictionaries: DictionaryProperties): void {
+        dictionaries.className = "Object";
+        dictionaries.containsKey = new NativeCallProperties(
+            "hasOwnProperty",
+            NativeCallScope.Member,
+            NativeCallType.Function);
+        dictionaries.keys = new NativeCallProperties(
+            "Object.keys",
+            NativeCallScope.Static,
+            NativeCallType.Function);
+        dictionaries.initializeEnd = "}";
+        dictionaries.initializePairComma = ",";
+        dictionaries.initializePairLeft = "";
+        dictionaries.initializePairMiddle = ": ";
+        dictionaries.initializePairRight = "";
+        dictionaries.initializeStart = "{";
+        dictionaries.typeLeft = "{ [i] ";
+        dictionaries.typeMiddle = "= ";
+        dictionaries.typeRight = " }";
+    }
+
+    /**
+     * Generates metadata on enums.
+     * 
+     * @param enums   A property container for metadata on enums.
+     */
+    protected generateEnumProperties(enums: EnumProperties): void {
+        super.generateEnumProperties(enums);
+
+        enums.declareStartLeft = "var ";

```

Comment:
`var` is evil! `var` is evil! `var` is evil!

`let` is our savior. (`const` too, but we can ignore it for now)